tests/tests/operation_tests.cpp

```diff
@@ -7423,7 +7424,450 @@ BOOST_FIXTURE_TEST_SUITE(auction_window_tests, votes_extended_fixture)
+                rwd += account.curation_rewards;
+            }
+            auto& alice_acc = db->get_account(alice_post.author);
+
+            double auw_tokens = (gpo.total_reward_fund_steem.amount.value);
+            double total_payout = (alice_acc.posting_rewards.value) * 100.0 / 75.0;
+            double voters_reward = (rwd.value);
+            double voters_reward_percent = (auw_tokens + voters_reward) / total_payout;
+            double modeled_voters_reward_percent = (total_comment_fund.reward_fund().amount.value + voters_reward) / total_payout;
+
+            double allowed_percent_delta = 0.01; // = 1e-2
+            double allowed_tokens_delta = 50;
+
+            BOOST_CHECK_EQUAL(fabs(0.25 - voters_reward_percent) < allowed_percent_delta, true);
+            BOOST_CHECK_EQUAL(abs(auw_tokens - total_comment_fund.reward_fund().amount.value) < allowed_tokens_delta, true);
+            BOOST_CHECK_EQUAL(fabs(0.25 - modeled_voters_reward_percent) < allowed_percent_delta, true);
+        }
+        FC_LOG_AND_RETHROW()
+    }
+
+    BOOST_AUTO_TEST_CASE(auction_window_tokens_to_reward_fund_empty_case) {
+        try {
+            BOOST_TEST_MESSAGE("Auction window reward goes to reward fund.");
+
+            // Needed to be sured, that auction window's been already enabled.
+            generate_blocks(fc::time_point_sec(STEEMIT_HARDFORK_0_6_REVERSE_AUCTION_TIME), true);
+
+            auto &wso = db->get_witness_schedule_object();
+            db->modify(wso, [&](witness_schedule_object &w) {
+                w.median_props.auction_window_size = 10 * 60;
+            });
+
+            generate_block();
+
+            ACTOR(alice);
+            int voters_count = 10;
+
+            comment_operation comment;
+            comment_options_operation cop;
+            signed_transaction tx;
+            set_price_feed(price(ASSET("1.000 GOLOS"), ASSET("1.000 GBG")));
+
+            BOOST_TEST_MESSAGE("Create a post.");
+            comment.author = "alice";
+            comment.permlink = "testauw2";
+            comment.parent_permlink = "test";
+            comment.title = "test";
+            comment.body = "Let's test auction window improvements!";
+
+            tx.operations.clear();
+            tx.signatures.clear();
+            GOLOS_CHECK_NO_THROW(push_tx_with_ops(tx, alice_private_key, comment));
+            generate_block();
+
+            protocol::comment_auction_window_reward_destination dest;
+            dest.destination = protocol::to_reward_fund;
+
+            cop.author = comment.author;
+            cop.permlink = comment.permlink;
+
+            cop.extensions.insert(dest);
+
+            tx.operations.clear();
+            tx.signatures.clear();
+
+            GOLOS_CHECK_NO_THROW(push_tx_with_ops(tx, alice_private_key, cop));
+            generate_block();
+            auto & alice_post = db->get<comment_object, by_permlink>(boost::make_tuple(comment.author, comment.permlink));
```

Comment:
1. auto & variable_name - Where do you find this strange code style? :)
It is bad, same as `auto &variable_name`
Good is `auto& variable_name`
And same in another places.
2. std::make_tuple is better than boost::